src/header/launchbar/BlastLaunchbarButton.tsx

```diff
@@ -0,0 +1,66 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useRef } from 'react';
+import classNames from 'classnames';
+
+import { useAppSelector } from 'src/store';
+
+import { getUnviewedBlastSubmissions } from 'src/content/app/tools/blast/state/blast-results/blastResultsSelectors';
+
+import LaunchbarButton from './LaunchbarButton';
+import { BlastIcon } from 'src/shared/components/app-icon';
+
+import styles from './Launchbar.scss';
+
+const BlastLaunchbarButton = () => {
+  const unviewedSubmissions = useAppSelector(getUnviewedBlastSubmissions);
+  const isAnyJobRunning = useRef(false);
+
+  isAnyJobRunning.current = unviewedSubmissions
+    .flatMap((submission) =>
+      submission.results.map((job) => job.status === 'RUNNING')
+    )
+    .some(Boolean);
+
+  const getStatusClass = () =>
+    classNames(
+      styles.jobStatus,
+      isAnyJobRunning.current
+        ? styles.jobStatusRunning
+        : styles.jobStatusFinished
+    );
+
+  const WrappedBlastIcon = () => (
+    <div className={styles.toolsIconWrapper}>
+      <BlastIcon />
+      {!!unviewedSubmissions.length && <div className={getStatusClass()}></div>}
+    </div>
+  );
```

Comment:
Several things here.

For one, we are practically guaranteed to have a similar notification system for other tools buttons in the launchbar (check with Andrea if in doubt); so we might benefit from a more general component that knows how to display a dot (`LaunchbarButtonWithNotification` or a shorter name if you can think of one).

Second, the `LaunchbarButton` component doesn't work with BLAST. Currently, `LaunchbarButton` can only create a single link to `/blast`; but we know that this is insufficient. The most likely behaviour that we will need is for the blast button to keep track which blast view the user was on last, and to take the user to that view when he clicks on the button. I don't know if this should be considered part of this PR; but it suggests that we might rethink the interface of the `LaunchbarButton` component.

Something for us to discuss, perhaps?

Reply:
✅